packages/types/src/translators/README.md

```diff
@@ -0,0 +1,25 @@
+# Translators
+
+This directory contains translators: pure functions that translate an object into a different object.
+
+## Organization
+
+Files are named and organized by the type of **input** that the translator accepts, not by the type of **output** that the translator returns. For example, a file with `TransactionPlan` -> `Transaction` and `TransactionPlan` -> `TransactionView` translators should be in `transaction-plan.ts`.
+
+## Naming
+
+Translator functions should start with the word `as` and end with the type which they are returning. For example, a translator that takes a `MemoView` and returns an opaque `MemoView` should be called `asOpaqueMemoView`. That will prevent naming collisions and make each function's purpose clearer when you're importing multiple translators at once:
+
+```TS
+import { asOpaqueAddressView, asOpaqueMemoView } from '@penumbra-zone/types';
+```
+
+## Delegation
+
+Translators should be as simple as possible, delegating to other translators for nested object types. For example, the `asPublicTransactionView` translator doesn't itself modify a transaction view's nested memo view; rather, it delegates that work to `asOpaqueMemoView`. This way, each translator is easy to understand, and easy to test.
+
+## Higher-order functions
+
+Some "translators" are actually higher-order functions (HOFs) that _return_ translators. This is a useful pattern for when a translator needs access to additional variables besides the translatable object.
```

Comment:
Curious what reviewers think of this design decision. The more I think about it, the more I'm not crazy about it — i.e., perhaps instead of HOFs, I should just pass a second `ctx` argument to translators which can hold whatever values they need...?

Reply:
I found them to be a bit harder to mentally grasp. Perhaps it could be the double parenthesis thing. If we find that we only use it like so `asReceiverTransactionView(isControlledAddress)(transactionView)`, that probably means we don't need the HOF right? Otherwise we'd be using just `asReceiverTransactionView(isControlledAddress)` more often right?

Reply:
The idea is that I'd like to have a standalone function that can be passed to e.g. `.map()`, like here: https://github.com/penumbra-zone/web/pull/387/files/c92b75e6b1b200de7ee7ffc52db1113625f07a34#diff-911daf37f54f94c28a4b25b6b2e923e5b45a3bf651ed7befef1da45f1d581ef7R32

That's only possible using HOFs . Otherwise, `.map()` needs to have an inline function defined.

That said, I agree that the HOF approach is harder to grasp, and that almost certainly outweighs the "need" for avoiding inline functions. I'll rewrite this with a `ctx` object that holds `isControlledAddress`.

Reply:
OK, I rewrote the translators to take a `ctx` object instead of using HOFs.